sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -44,14 +63,22 @@ case class ArrowEvalPythonExec(udfs: Seq[PythonUDF], output: Seq[Attribute], chi
     val schemaOut = StructType.fromAttributes(output.drop(child.output.length).zipWithIndex
       .map { case (attr, i) => attr.withName(s"_$i") })
 
+    val batchSize = conf.arrowMaxRecordsPerBatch
+
+    val batchIter = if (batchSize > 0) {
+      new BatchIterator(iter, batchSize)
+    } else {
+      Iterator(iter)
+    }
```

Comment:
This could be done on one line, not sure if others agree with that style though

Reply:
I changed to one line. I think more concise.